### ISSUE TITLE
[Action Popover] Focus shift not announced to screen reader

### ISSUE THREAD

**Issue by Fenwick17** [Sage/carbon#7790]
### Description

When using the Action Popover with VoiceOver and a user expands the button, the first button within is not announced. And they are then only announced through navigation which would skip the first one.
This causes additional issues when the action popover only has one button within it, as then the user can never access that button.

There are mutliple options available to fix this. 
- One would be to just mirror behaviour of the 'Multi-action button' and 'Split button' so that focus does not shift, and a user is just informed it has expanded. 
- Blur the original focus before applying the new focus

### Reproduction

https://carbon.sage.com/?path=/story/action-popover--default

### Steps to reproduce

Expand the action popover on VoiceOver

### Carbon version

158.32.0

### React version

v18

### Relevant browsers

Safari

### Relevant OSs

MacOS

### Assistive technologies

VoiceOver

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.
